src/components/StageTransitions.js

```diff
@@ -84,26 +88,52 @@ export class Transition extends React.Component {
       this.props.fetchingStart();
       const response = await axios.get(this.transitionURL, {});
       let newData;
+      let joinedStudent = [];
+      let joinedOutreach = [];
+      const campusMilestoneKey = Object.keys(campusStageOfLearning);
+
       if (this.props.loggedInUser) {
-        newData = response.data.data.map((v) => ({
-          ...v,
-          loggedInUser: this.props.loggedInUser,
-        }));
+        newData = response.data.data.map((v) => {
+          if (campusMilestoneKey.indexOf(v.to_stage) !== -1) {
+            joinedStudent.push(v);
+          } else {
+            joinedOutreach.push(v);
+          }
+          return {
+            ...v,
+            loggedInUser: this.props.loggedInUser,
+          };
+        });
       } else {
-        newData = response.data.data.map((v) => ({
-          ...v,
-        }));
+        newData = response.data.data.map((v) => {
+          if (campusMilestoneKey.indexOf(v.to_stage) !== -1) {
+            joinedStudent.push(v);
+          } else {
+            joinedOutreach.push(v);
+          }
+          return {
+            ...v,
+          };
+        });
+      }
```

Comment:
From a purist standpoint, I wouldn't modify an existing Array in the callback function to `map` as the intended use of this functional programming construct is to create a new output Array from an existing input one by running each item through a function without side effects (e.g., `const doubledArr = arr.map(x => 2*x)`).  `forEach` would be more appropriate for an imperative approach when you're giving commands such as add `v` to the end of the `joinedStudent` or `joinedOutreach` `Array` via `push`.

However, if you added this to `map` to only iterate through the `Array` once, you could avoid the code duplication here as follows:

```
const userInfo = this.props.loggedInUser ?
    {loggedInUser: this.props.loggedInUser} :
    {};
newData = response.data.data.map(v => {
  if (campusMilestoneKey.indexOf(v.to_stage) !== -1) {
    joinedStudent.push(v);
  } else {
    joinedOutreach.push(v);
  }  
  return {
    ...v,
    ...userInfo
  };
});
```

Alternatively, you could avoid changing the `newData` assignment and use `filter` instead if you don't mind iterating through `response.data.data` a couple more times as follows:

```
const joinedStudent = response.data.data.filter(v => campusMilestoneKey.indexOf(v.to_stage) !== -1);
const joinedOutreach = response.data.data.filter(v => campusMilestoneKey.indexOf(v.to_stage) === -1);
```

You could also use `includes` for slightly increased readability:

```
const joinedStudent = response.data.data.filter(v => campusMilestoneKey.includes(v.to_stage));
const joinedOutreach = response.data.data.filter(v => !campusMilestoneKey.includes(v.to_stage));
```

But perhaps this partition of the `user` Array into two Arrays `joinedStudent` and `joinedOutreach` to be stored in state properties is not necessary?  Could you instead set a local variable `filteredData` in the render method, which stores the current one that you want to show, and then pass it as the `data` prop to `MUIDataTable` as follows:

```
let filteredData = this.state.data;
if(campusPath === 'campus') {
  filteredData = this.state.toggleOutreach ?
      filteredData.filter(v => !campusMilestoneKey.includes(v.to_stage)) :
      filteredData.filter(v => campusMilestoneKey.includes(v.to_stage))
}

<MUIDataTable
    columns={StudentService.columns[this.props.dataType]}
    data={filteredData}
```

The change listener could then just toggle the state of `toggleOutreach`:

```
this.setState({toggleOutreach: !this.state.toggleOutreach});
```

Of course, you'd be recomputing the `filteredData` Array on each toggle/re-render, but I don't think this would cause a major performance hit.  And if for some reason it did, you could still determine it in the render method and cache the result as needed (e.g., by resetting `this.state.joinedStudent` and `this.state.joinedOutreach` to the empty Arrays each time you get new data to indicate that it needs to be recomputed here.  This assumes that there would be at least one student in each category.)